packages/compass-crud/src/components/crud-toolbar.tsx

```diff
@@ -0,0 +1,238 @@
+          store: localAppRegistry.getStore(queryBarRole.storeName!),
+          actions: localAppRegistry.getAction(queryBarRole.actionName!),
+        }
+      : null
+  );
+
+  const displayedDocumentCount = useMemo(
+    () => (loadingCount ? '' : `${count ?? 'N/A'}`),
+    [loadingCount, count]
+  );
+
+  const onClickRefreshDocuments = useCallback(() => {
+    track('Query Results Refreshed');
+    refreshDocuments();
+  }, [refreshDocuments]);
+
+  const QueryBarComponent = isExportable
+    ? queryBarRef.current!.component
+    : null;
+
+  const controlId = useId();
+  const prevButtonDisabled = useMemo(() => page === 0, [page]);
+  const nextButtonDisabled = useMemo(
+    () => (count ? 20 * (page + 1) >= count : false),
+    [count, page]
+  );
+
+  return (
+    <Toolbar className={crudToolbarStyles}>
+      <div className={crudQueryBarStyles}>
+        {isExportable && QueryBarComponent && (
+          <QueryBarComponent
+            store={queryBarRef.current!.store}
+            actions={queryBarRef.current!.actions}
+            resultId={resultId}
+            buttonLabel="Find"
+            onApply={onApplyClicked}
+            onReset={onResetClicked}
+          />
+        )}
+      </div>
+      <div className={crudBarStyles}>
+        <div className={toolbarLeftActionStyles}>
+          {!readonly && (
+            <AddDataMenu
+              insertDataHandler={insertDataHandler}
+              isWritable={isWritable}
+              instanceDescription={instanceDescription}
+            />
+          )}
+          <Button
+            leftGlyph={<Icon glyph="Export" />}
+            data-testid="export-collection-button"
+            size="xsmall"
+            onClick={openExportFileDialog}
+          >
+            Export Collection
+          </Button>
+        </div>
+        <div className={toolbarRightActionStyles}>
+          <Body data-testid="crud-document-count-display">
+            {start} - {end} of {displayedDocumentCount}
+          </Body>
+          {loadingCount && (
+            <SpinLoader size="12px" title="Fetching document count…" />
+          )}
+          {!loadingCount && (
+            <IconButton
+              aria-label="Refresh documents"
+              title="Refresh documents"
+              data-testid="refresh-documents-button"
+              onClick={onClickRefreshDocuments}
+            >
+              <Icon glyph="Refresh" />
+            </IconButton>
+          )}
+
+          <div>
+            <IconButton
+              data-testid="docs-toolbar-prev-page-btn"
+              aria-label="Previous Page"
+              title="Previous Page"
+              onClick={() => getPage(page - 1)}
+              disabled={prevButtonDisabled}
+            >
+              <Icon glyph="ChevronLeft" />
+            </IconButton>
+            <IconButton
+              data-testid="docs-toolbar-next-page-btn"
+              aria-label="Previous Page"
+              title="Previous Page"
```

Comment:
Both these buttons say Previous Page, btw. This one should probably be Next Page.